### PR TITLE
Index page sorting

### PR DESCRIPTION
**Pull request by n-flint**
Implemented three search links in the projects controller index action. The projects are ordered by date by default, there is a link to sort alphabetically, and a link to see projects with no drivers attached to them yet.

The project#index could probably use some refactoring, I am open to ideas. Everything works for now, but it is a weird looking conditional.  

link to story https://www.pivotaltracker.com/story/show/166161932

### CODE REVIEW

**Comment by n-flint** [comment]
Also, used the 'orderly' gem to test the order of elements appearing in the view. So you may have to bundle etc.

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+    @projects = Project.where(active: true).order(:date)
+    if params[:sort_by] == 'a-z'
+      @projects = Project.where(active: true).order(:title)
+    elsif params[:sort_by] == 'needs_driver'
+      needs_driver = @projects.find_all do |project|
+        project.carpools.empty?
+      end
+      @projects = needs_driver 
+    else
+    end
```
Comment (joequincy) [inline app/controllers/projects_controller.rb]: We could probably move this a class method on the Project model, which would take `params[:sort_by]` as an argument and return the sorted projects.

That way, we could move the conditional logic out of the Controller and into the Model.

**Comment by n-flint** [comment]
Thanks Jon! I refactored using a class method .sorted() in the project model. Should be good to go.

**Comment by erin-king** [comment]
The method in the controller looks much cleaner- thank you for your suggestion, Jon, and thanks for tidying it up, Noah.

**Pull request merged**